src/lazy/text/buffer.rs

```diff
@@ -1001,10 +1006,41 @@ impl<'data> TextBufferView<'data> {
         Self::match_text_until_unescaped(self, b'\"')
     }
 
-    fn match_long_string(self) -> IonParseResult<'data, MatchedString> {
-        // TODO: implement long string matching
-        //       The `fail` parser is a nom builtin that never matches.
-        fail(self)
+    /// Matches a long string comprised of any number of `'''`-enclosed segments interleaved
+    /// with optional comments and whitespace.
+    pub fn match_long_string(self) -> IonParseResult<'data, MatchedString> {
```

Comment:
🗺️ I recommend going to look at the `MatchedString` enum before reading this implementation. It explains why we're tracking the number of segments and the presence of escaped characters.